src/Defines.h

```diff
@@ -425,6 +431,24 @@ extern eDamageType StringToDamageType(const AString & a_DamageString);
 If a_Inverse is true, the opposite direction is used instead. */
 void AddFaceDirection(int & a_BlockX, int & a_BlockY, int & a_BlockZ, eBlockFace a_BlockFace, bool a_bInverse = false);
 
+/** a_CurrentPosiiton - the block of whom we would like to get the block next to
+a_BlockFace - the face (direction) in which we would like to get the block
+*/
+inline Vector3i GetBlockNextTo(Vector3i a_CurrentPosition, eBlockFace a_BlockFace)
+{
+	switch (a_BlockFace)
+	{
+		case BLOCK_FACE_XM: return Vector3i(a_CurrentPosition.x - 1, a_CurrentPosition.y, a_CurrentPosition.z);
+		case BLOCK_FACE_XP: return Vector3i(a_CurrentPosition.x + 1, a_CurrentPosition.y, a_CurrentPosition.z); 
+		case BLOCK_FACE_YM: return Vector3i(a_CurrentPosition.x, a_CurrentPosition.y - 1, a_CurrentPosition.z); 
+		case BLOCK_FACE_YP: return Vector3i(a_CurrentPosition.x, a_CurrentPosition.y + 1, a_CurrentPosition.z); 
+		case BLOCK_FACE_ZM: return Vector3i(a_CurrentPosition.x, a_CurrentPosition.y, a_CurrentPosition.z - 1); 
+		case BLOCK_FACE_ZP: return Vector3i(a_CurrentPosition.x, a_CurrentPosition.y, a_CurrentPosition.z + 1);
+		default: return a_CurrentPosition;
+	}
+}
```

Comment:
Put the function implementation into the cpp file instead.
The doxy comment should first say what the function does.